### PR TITLE
Feature/users list

### PR DESCRIPTION
**Pull request by michalpokojski**
Pull request DONE edit user button with edit forn still need to be added

### CODE REVIEW

File: src/components/Users.js [hunk 1]
```diff
@@ -1,10 +1,107 @@
...
+  handleClose = () => {
+    this.setState({open: false})
+  }
+
+  handleCloseWithSubmit = () => {
+    this.props.handleSubmit()
+    setTimeout(() => this.setState({open: !this.props.submitStatus.submitSucceeded}), 1000)
+  }
   render() {
+    let data = sortByStringAscending([...users.concat(this.props.newUsers)], 'email')
```
Comment (bachbach) [inline src/components/Users.js]: There are a lot of logic. I guess it's filter feature? I propose to move it to container, to keep components stupid

File: src/containers/RemoveButton.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { connect } from 'react-redux'
+import RemoveButton from "../components/RemoveButton";
+import { removeUser } from '../reducers/filters'
+
```
Comment (bachbach) [inline src/containers/RemoveButton.js]: remove space

File: src/containers/RemoveButton.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+export default connect(
+  null,
+  dispatch => ({
+    remover: userEmail => dispatch(removeUser(userEmail))
```
Comment (bachbach) [inline src/containers/RemoveButton.js]: remover? what does it mean :D?

File: src/containers/NewUserForm.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { reduxForm } from "redux-form"
+import NewUserForm from '../components/NewUserForm'
+import submitNewUser from '../containers/submitNewUser'
+
```
Comment (bachbach) [inline src/containers/NewUserForm.js]: remove space

File: src/components/NewUserForm.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import MenuItem from 'material-ui/MenuItem'
+
+let defaultValue = 'user'
+
+const CustomSelectField = ({input, value, label, meta: {touched, error}, children, ...custom},) => (
+  <SelectField
+    floatingLabelText={label}
+    errorText={touched && error}
+    {...input}
+    onChange={(event, index, val) => {
```
[the remark this answers is not in the document]
Comment (michalpokojski) [reply src/components/NewUserForm.js]: already fixed this overwriting

File: src/containers/RemoveButton.js (hunk 2 above)
Comment (michalpokojski) [reply src/containers/RemoveButton.js]: removed

File: src/containers/NewUserForm.js (hunk 4 above)
Comment (michalpokojski) [reply src/containers/NewUserForm.js]: removed

File: src/components/Users.js [hunk 6]
```diff
@@ -1,10 +1,107 @@
 import React, { Component } from 'react'
+import DataTables from 'material-ui-datatables'
+import users from '../data/users.json'
+import { sortByStringAscending, sortByStringDescending } from '../helpers/sorting'
+import { TABLE_COLUMNS_USERS } from '../constants/tableColumnsSpecifications'
```
[the remark this answers is not in the document]
Comment (michalpokojski) [reply src/components/Users.js]: 👍

**Comment by michalpokojski** [comment]
And about this logic in ```Users.js``` i have no idea how would i make it.

**Comment by bachbach** [comment]
I have an idea. Please catch me in my free time. Hope it's possible :P

**Comment by bachbach** [comment]
Btw, otherwise everything is ok 👍

File: src/components/Users.js [hunk 7]
```diff
@@ -7,81 +7,74 @@ import RaisedButton from 'material-ui/RaisedButton'
...
 
-  handleClose = () => {
-    this.setState({openUserCreator: false})
-  }
+  handleClose = () => this.setState({openUserCreator: false})
 
-  handleCloseWithSubmit = () => {
-    this.props.handleSubmit()
-    setTimeout(() => this.setState({openUserCreator: !this.props.submitStatus.submitSucceeded}), 1000)
+  handleCloseWithSubmit = (newOrEdit) => () => {
```
Comment (bachbach) [inline src/components/Users.js]: I suggest to use in that case bool paramters. They are less than string. So in that case i suggest:
```
handleCloseWithSubmit = (isNew) => () => {
```

File: src/components/Users.js [hunk 8]
```diff
@@ -7,81 +7,74 @@ import RaisedButton from 'material-ui/RaisedButton'
...
+    newOrEdit === 'new' ? this.props.handleSubmitNewUser() : this.props.handleSubmitEditUser()
+    setTimeout(() => newOrEdit === 'new' ?
+      this.setState({openUserCreator: !this.props.submitStatusNewUser.submitSucceeded}) :
+      this.props.saveEditing(this.props.submitStatusEditUser.submitSucceeded), 100)
   }
+
+
   render() {
+    const editedEmails = this.props.editedUsers.map(user => user.email)
+    console.log(editedEmails)
```
Comment (bachbach) [inline src/components/Users.js]: remove console log

File: src/components/Users.js [hunk 9]
```diff
@@ -7,81 +7,74 @@ import RaisedButton from 'material-ui/RaisedButton'
...
       .filter(row =>
         row.email.includes(this.state.searchPhrase) ||
         row.lastName.includes(this.state.searchPhrase) ||
-        row.firstName.includes(this.state.searchPhrase))
+        row.firstName.includes(this.state.searchPhrase)
+      )
 
     let displayData = data.slice(this.state.rowSize * (this.state.page - 1), this.state.rowSize * (this.state.page))
-      .filter(row => !this.props.removedUsers.some(name => name === row.email))
+      .filter(row => !this.props.removedUsers.some(name => name === row.email)).filter(row => !editedEmails.some(name => name === row.email)).concat(this.props.editedUsers)
```
Comment (bachbach) [inline src/components/Users.js]: can't we use 1 filter?

File: src/components/Users.js [hunk 10]
```diff
@@ -7,81 +7,74 @@ import RaisedButton from 'material-ui/RaisedButton'
...
         label="Cancel"
-        primary={true}
+        secondary
         onClick={this.handleClose}
       />,
       <FlatButton
         label="Submit"
         primary={true}
-        onClick={this.handleCloseWithSubmit}
+        onClick={this.handleCloseWithSubmit('new')}
```
Comment (bachbach) [inline src/components/Users.js]: so, regarding to my previous comment, here you can use `onClick={this.handleCloseWithSubmit(true)}`

File: src/containers/EditUserForm.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import { reduxForm } from "redux-form"
+import EditUserForm from '../components/EditUserForm'
+import submitEditedUser from '../containers/submitEditedUser'
+
```
Comment (bachbach) [inline src/containers/EditUserForm.js]: remove eneter

File: src/containers/submitEditedUser.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import { submitEditedUser } from '../reducers/usersOperations'
+
+const delayThisBy = ms => new Promise(resolve => setTimeout(resolve, ms))
+
+function submitNewUser(values, dispatch, props) {
+  console.log(props)
```
Comment (bachbach) [inline src/containers/submitEditedUser.js]: remove console

File: src/containers/submitEditedUser.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+  return delayThisBy(1).then(() => dispatch(submitEditedUser({
+      email: props.userData.email,
```
Comment (bachbach) [inline src/containers/submitEditedUser.js]: Please check some merge function to handle this case

File: src/reducers/usersOperations.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
+const ADD_USER = 'addNewUser/ADD_USER'
+const EDIT_USER = 'addNewUser/EDIT_USER'
+const CURRENTLY_EDIT_USER = 'userEdit/CURRENTLY_EDIT_USER'
```
Comment (bachbach) [inline src/reducers/usersOperations.js]: what is difference between `EDIT_USER ` and `CURRENTLY_EDIT_USER ` ?

**Comment by bachbach** [comment]
I guess we can skip for now refactor of logic. But hope in near feature we will have some time to check it :)

**Pull request merged**